libs/utils/utils.js

```diff
@@ -863,6 +904,84 @@ function decorateMeta() {
   });
 }
 
+function decorateDocumentExtras(config) {
+  decorateMeta();
+  decorateHeader();
+  decorateFooterPromo(config);
+
+  import('./samplerum.js').then(({ addRumListeners }) => {
+    addRumListeners();
+  });
+}
+
+async function documentPostSectionLoading(config) {
+  const georouting = getMetadata('georouting') || config.geoRouting;
+  if (georouting === 'on') {
+    // eslint-disable-next-line import/no-cycle
+    const { default: loadGeoRouting } = await import('../features/georoutingv2/georoutingv2.js');
+    loadGeoRouting(config, createTag, getMetadata, loadBlock, loadStyle);
+  }
+  const appendage = getMetadata('title-append');
+  if (appendage) {
+    import('../features/title-append/title-append.js').then((module) => module.default(appendage));
+  }
+  if (getMetadata('seotech-structured-data') === 'on' || getMetadata('seotech-video-url')) {
+    import('../features/seotech/seotech.js').then((module) => module.default(
+      { locationUrl: window.location.href, getMetadata, createTag, getConfig },
+    ));
+  }
+  const richResults = getMetadata('richresults');
+  if (richResults) {
+    const { default: addRichResults } = await import('../features/richresults.js');
+    addRichResults(richResults, { createTag, getMetadata });
+  }
+  loadFooter();
+  const { default: loadFavIcon } = await import('./favicon.js');
+  loadFavIcon(createTag, getConfig(), getMetadata);
+  if (config.experiment?.selectedVariant?.scripts?.length) {
+    config.experiment.selectedVariant.scripts.forEach((script) => loadScript(script));
+  }
+  initSidekick();
+
+  const { default: delayed } = await import('../scripts/delayed.js');
+  delayed([getConfig, getMetadata, loadScript, loadStyle, loadIms]);
+}
+
+async function processSection(section, config, isDoc) {
+  const inlineFrags = [...section.el.querySelectorAll('a[href*="#_inline"]')];
+  if (inlineFrags.length) {
+    const { default: loadInlineFrags } = await import('../blocks/fragment/fragment.js');
+    const fragPromises = inlineFrags.map((link) => loadInlineFrags(link));
+    await Promise.all(fragPromises);
+    const newlyDecoratedSection = decorateSection(section.el, section.idx);
```

Comment:
ouch - Could we not refactor this entire `if` block to `decorateSection`, to prevent this rerun of `decorateSection`?

Reply:
No we have to run it again:
1. Inline fragment(s) found
2. Download all of the fragments and insert them into the DOM as if they were part of the original page
3. Now DOM has changed, but that new DOM hasn't been processed
4. Load `decorateSection` again to decorate the updated DOM

However if you see a better way to do this please lmk